### PR TITLE
Personal video module

### PR DESCRIPTION
**Pull request by ccfredbaum**

### CODE REVIEW

File: my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		add_action( 'mvr_module_submenu_add', array( $this, 'personal_meetingroom_menu_button' ) );
+	}
+	/**
+	 * Render Module Menu.
+	 */
+	public function personal_meetingroom_menu_button() {
+		$name = self::MODULE_PERSONAL_MEETING_DISPLAY;
+		$slug = self::MODULE_PERSONAL_MEETING_NAME;
+		//phpcs:ignore --WordPress.WP.I18n.NonSingularStringLiteralText - $name is a constant text literal already.
+		$display = esc_html__( $name, 'myvideoroom' );
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php:91]: `esc_html( $name, 'myvideoroom' );`

No need to call the translate function here - as it won't do anything.

File: my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		echo '<a class="mvr-menu-header-item" href="?page=my-video-room-extras&tab=' . esc_html( $slug ) . '">' . esc_html( $display ) . '</a>';
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php:92]: `esc_html( $display )`

You're also now escaping it twice

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	ob_start();
+
+	?>
+	<div class="mvr-outer-box-wrap">
+
+	<h1 class="mvr-heading-head"><?php esc_html_e( 'Personal Meeting Rooms', 'my-video-room' ); ?></h1>
+
+		<p>
+		<?php
+		esc_html_e(
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:35]: multi-line translation, likely to cause some problems

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+			<?php
+			//phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Function already escaped above.
+			echo factory::get_instance( MVRSiteVideoViews::class )->generate_room_table( MVRPersonalMeeting::MODULE_PERSONAL_MEETING_NAME );
+			?>
+
+			<h2><?php esc_html_e( 'Customizing the Room', 'my-video-room' ); ?></h2>
+			<p><?php esc_html_e( 'You can edit your room look and feel with any page editor of your choice - the page must contain the shortcode ', 'my-video-room' ); ?> <strong> [ccmeetswitch]</strong>	</p>
+			<p>
+			<?php
+			esc_html_e(
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:60]: multi-line translation here again

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	</div>
+	<div class="wrap">
+		<h2><?php esc_html_e( 'Current Room Information', 'my-video-room' ); ?></h2>
+			<?php
+			//phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Function already escaped above.
+			echo factory::get_instance( MVRSiteVideoViews::class )->generate_room_table( MVRPersonalMeeting::MODULE_PERSONAL_MEETING_NAME );
+			?>
+
+			<h2><?php esc_html_e( 'Customizing the Room', 'my-video-room' ); ?></h2>
+			<p><?php esc_html_e( 'You can edit your room look and feel with any page editor of your choice - the page must contain the shortcode ', 'my-video-room' ); ?> <strong> [ccmeetswitch]</strong>	</p>
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:57]: is `[ccmeetswitch]` the correct shortcode?

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+		// Activation/module.
+		Factory::get_instance( ModuleConfig::class )->module_activation_button( MVRPersonalMeeting::MODULE_PERSONAL_MEETING_ID );
+		?>
+
+	</div>
+	<div class="wrap">
+		<h2><?php esc_html_e( 'Current Room Information', 'my-video-room' ); ?></h2>
+			<?php
+			//phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Function already escaped above.
+			echo factory::get_instance( MVRSiteVideoViews::class )->generate_room_table( MVRPersonalMeeting::MODULE_PERSONAL_MEETING_NAME );
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:53]: `Factory` should have a capital `F` - may be ok now, but likely to cause bugs in future

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+				'my-video-room'
+			);
+			?>
+			</p>
+			<p><strong><?php esc_html_e( 'Note', 'my-video-room' ); ?></strong><?php esc_html_e( '- This setting is shared with BuddyPress personal profile room integration as they are the same room.', 'my-video-room' ); ?> </p>
+					<?php
+						$layout_setting = Factory::get_instance( UserVideoPreference::class )->choose_settings(
+							SiteDefaults::USER_ID_SITE_DEFAULTS,
+							MVRPersonalMeeting::ROOM_NAME_PERSONAL_MEETING,
+							array( 'basic', 'premium' )
```
Comment (alecsammon) [inline my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:94]: i don't think this array parameter is used anymore down the chain

File: my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php [hunk 8]
```diff
@@ -132,7 +134,7 @@ private function get_sitevideo_admin_page(): string {
 	 *
 	 * @return string
 	 */
-	public function create_site_conference_page(): string {
+	public function create_site_conference_page( bool $shortcode = null ): string {
```
Comment (alecsammon) [inline my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php:137]: missing comment for `$shortcode` in the docblock

File: my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php [hunk 9]
```diff
@@ -246,6 +250,23 @@ function ( $room_id ) {
...
+	 * Get the list of current rooms
+	 *
+	 * @param string    $room_type     Category of Room if used.
+	 * @param \stdClass $room The room object.
+	 *
+	 * @return ?string
+	 */
+	public function conference_check_reception_status( string $room_type = null, \stdClass $room ): ?string {
+		$room_name   = Factory::get_instance( SiteDefaults::class )->room_map( 'sitevideo', $room->display_name );
+		$text_single = 'One Guest Waiting';
```
Comment (alecsammon) [inline my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php:263]: do these need translated?

File: my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php (hunk 1 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php:91]: Fixed

File: my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php (hunk 2 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/class-mvrpersonalmeeting.php:92]: Fixed

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php (hunk 3 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:35]: Fixed

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php (hunk 4 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:60]: Fixed

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php (hunk 5 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:57]: Good catch - corrected.

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php (hunk 6 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:53]: Fixed

File: my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php (hunk 7 above)
Comment (ccfredbaum) [reply my-video-room/module/personalmeetingrooms/views/view-settings-personalvideo.php:94]: Removed

File: my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php (hunk 8 above)
Comment (ccfredbaum) [reply my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php:137]: Done

File: my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php (hunk 9 above)
Comment (ccfredbaum) [reply my-video-room/module/sitevideo/library/class-mvrsitevideoroomhelpers.php:263]: Yes- translation added

**Review by astroyouth — APPROVED** [review]

**Pull request merged**